pkg/clusteragent/admission/patch/patcher_test.go

```diff
@@ -13,119 +13,102 @@ import (
-	req.Action = EnableConfig
-	req.Revision = 12345
-	req.LibConfig = common.LibConfig{Language: "java", Version: "latest", TracingTags: []string{"k1:v1", "k2:v2"}}
-	require.NoError(t, p.patchDeployment(req))
+	expectedLabels = map[string]string{k8sutil.RcLabelKey: "false"}
+	requireMapHas(t, expectedLabels, got.ObjectMeta.Labels)
+	requireMapHas(t, expectedAnnotations, got.ObjectMeta.Annotations)
 
-	// Check the new patch
-	got, err = client.AppsV1().Deployments(ns).Get(context.TODO(), name, metav1.GetOptions{})
-	require.NoError(t, err)
-	require.Equal(t, got.Spec.Template.Labels["admission.datadoghq.com/enabled"], "true")
-	require.Equal(t, got.Spec.Template.Annotations["admission.datadoghq.com/java-lib.version"], "latest")
-	require.Equal(t, got.Spec.Template.Annotations["admission.datadoghq.com/java-lib.config.v1"], `{"library_language":"java","library_version":"latest","tracing_tags":["k1:v1","k2:v2"]}`)
-	require.Equal(t, got.Spec.Template.Annotations["admission.datadoghq.com/rc.id"], "id")
-	require.Equal(t, got.Spec.Template.Annotations["admission.datadoghq.com/rc.rev"], "12345")
-	require.Equal(t, got.Annotations["admission.datadoghq.com/rc.id"], "id")
-	require.Equal(t, got.Annotations["admission.datadoghq.com/rc.rev"], "12345")
-
-	// Stage a new patch with a new config
-	req.Action = StageConfig
-	req.Revision = 123456
-	req.LibConfig = common.LibConfig{Language: "java", Version: "v1", TracingTags: []string{"foo:bar"}}
-	require.NoError(t, p.patchDeployment(req))
-
-	// Check the new patch, only the deployment annotations should be updated
-	got, err = client.AppsV1().Deployments(ns).Get(context.TODO(), name, metav1.GetOptions{})
+	// Delete configuration
+	req.Action = DeleteConfig
+	req.Revision = 15
+	require.NoError(t, p.patchNamespaces(req))
+
+	// Check the patch
+	got, err = client.CoreV1().Namespaces().Get(context.TODO(), name, metav1.GetOptions{})
 	require.NoError(t, err)
-	require.Equal(t, got.Spec.Template.Labels["admission.datadoghq.com/enabled"], "true")
-	require.Equal(t, got.Spec.Template.Annotations["admission.datadoghq.com/java-lib.version"], "latest")
-	require.Equal(t, got.Spec.Template.Annotations["admission.datadoghq.com/java-lib.config.v1"], `{"library_language":"java","library_version":"latest","tracing_tags":["k1:v1","k2:v2"]}`)
-	require.Equal(t, got.Spec.Template.Annotations["admission.datadoghq.com/rc.id"], "id")
-	require.Equal(t, got.Spec.Template.Annotations["admission.datadoghq.com/rc.rev"], "12345")
-	require.Equal(t, got.Annotations["admission.datadoghq.com/rc.id"], "id")
-	require.Equal(t, got.Annotations["admission.datadoghq.com/rc.rev"], "123456")
+	require.NotContains(t, got.ObjectMeta.Labels, k8sutil.RcLabelKey)
+	require.NotContains(t, got.ObjectMeta.Annotations, k8sutil.RcIDAnnotKey)
+	require.NotContains(t, got.ObjectMeta.Annotations, k8sutil.RcRevisionAnnotKey)
+}
+
+func requireMapHas(t *testing.T, expected map[string]string, actual map[string]string) {
```

Comment:
Should we just compare the maps using `reflect.DeepEqual` to avoid defining our own functions?